src/bionics.cpp

```diff
@@ -204,21 +204,18 @@ bionic_data const& bionic_info(std::string const &id)
     debugmsg("bad bionic id");
 
     static bionic_data const null_value {"bad bionic", false, false, 0, 0, 0, 0, 0,
-                                         "bad_bionic", false,
-                                         { num_bp , 1 } };
+                                         "bad_bionic", false };
     return null_value;
 }
 
 void bionics_install_failure( player *u, int difficulty, int success,
-                              std::map<body_part, int> affected_bodyparts );
+                              std::map<body_part, size_t> affected_bodyparts );
 
 bionic_data::bionic_data(std::string nname, bool ps, bool tog, int pac, int pad, int pot,
-                         int ct, int cap, std::string desc, bool fault,
-                         std::map<body_part, int> aff_bps
+                         int ct, int cap, std::string desc, bool fault
 ) : name(std::move(nname)), description(std::move(desc)), power_activate(pac),
```

Comment:
> std::move  is probably a no-op here.

The compiler may replace the copy construction with a move construction, but only if it can prove this does not change the behavior of the code.

>  If it isn't there's a related problem as you haven't given the caller any indication you intend to steal their string.

The parameter is by-value, the caller copies the string on their own, there is nothing to steal from.

> You could try  const std::string& desc  as a parameter and then also possibly specify  const std::string description  in the header presuming there's no code that needs to change the description.

Don't use `const &` here. The by-value is ideal. If it's called with a string literal, that literal is used to create a temporary, directly in the place of the argument. It is than moved out of into the member. Effectively one construction from literal and one move construction.

A const reference would require a copy from the argument into the member. Effectively one construction from literal and one *copy* construction.

----

Rule of thumb: if you're going to move the argument somewhere, take it as value, not const reference.